xdsl/dialects/gpu.py

```diff
@@ -488,6 +502,82 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class LaunchFuncOp(IRDLOperation):
+    """
+    Launch a kernel function on the specified grid of thread blocks. gpu.launch
+    operations are lowered to gpu.launch_func operations by outlining the kernel body
+    into a function in a dedicated module, which reflects the separate compilation
+    process. The kernel function is required to have the gpu.kernel attribute. The
+    module containing the kernel function is required to be a gpu.module. And finally,
+    the module containing the kernel module (which thus cannot be the top-level module)
+    is required to have the gpu.container_module attribute. The gpu.launch_func operation
+    has a symbol attribute named kernel to identify the fully specified kernel function
+    to launch (both the gpu.module and func).
+
+    The gpu.launch_func supports async dependencies: the kernel does not start executing
+    until the ops producing those async dependencies have completed.
+
+    By the default, the host implicitly blocks until kernel execution has completed. If
```

Comment:
By default

Reply:
Oh wow! I literally copy pasted the MLIR doc and did not read totally through. You found a mistake in MLIR's doc 🙂